File: frontend/src/features/owner-room-details/types/roomHouseholderDetailDto.ts
export interface BookingDto {
	id: number;
	bookerId: string;
	bookerName: string;
	bookerEmail: string;
	bookerPhoneNumber: string;
	bookingStatus: string;
	roomId: number;
	bookerImageUrl: string;
}

export interface RoomImageInfo {
	id: number;
	url: string;
}

export interface RoomHouseholderDetailDto {
	id: number;
	name: string;
	latitude: number;
	longitude: number;
	description: string;
	price: number;
	roomStatus: string;
	imageRoomUrls: string[];
	images: RoomImageInfo[];
	services: string[];
	policies: RoomPolicyDto[];
	bookings: BookingDto[];
}

export interface RoomPolicyDto {
	code: string;
	description: string;
}

File: backend/src/HousingApp.Application/Room/DTOs/RoomHouseholderDetailDto.cs
namespace HousingApp.Application.Room.DTOs;

public record RoomHouseholderDetailDto(
    int Id,
    string Name,
    double Latitude,
    double Longitude,
    string Description,
    decimal Price,
    string RoomStatus,
    List<string> ImageRoomUrls,
    List<RoomImageDto> Images,
    List<string> Services,
    List<RoomPolicyDto> Policies,
    List<BookingDto> Bookings
);
